docs: contributing guide should include pre-PR checklist

## Summary\nThe contributor guide documents setup and linting, but it still omits the repo-standard pre-PR checks and issue-linking expectations. That leaves contributors with an incomplete workflow relative to the current PR template and agent skills.\n\n## Proposed change\nUpdate  so it tells contributors to run   Ruff linter............................. ok
  Black formatter (lib/levanter/pyproject.toml) ok
  License headers (lib/levanter/etc/license_header.txt) ok
  Ruff linter............................. ok
  Black formatter (lib/haliax/pyproject.toml) ok
  License headers (lib/haliax/etc/license_header.txt) ok
  Ruff linter............................. ok
  Black formatter......................... ok
  License headers (etc/license_header.txt) ok
  Pyrefly type checker.................... ok
  Large files............................. ok
  Python AST.............................. ok
  Merge conflicts......................... ok
  TOML and YAML........................... ok
  Trailing whitespace..................... ok
  End-of-file newline..................... ok
  Jupyter notebooks....................... ok
============================================================
OK
============================================================, start with targeted ============================= test session starts ==============================
platform linux -- Python 3.11.14, pytest-8.4.2, pluggy-1.6.0
rootdir: /home/golem/oa-jder-bot/worktrees/github.com/marin-community/marin/5979813734440518336
configfile: pyproject.toml
testpaths: tests, experiments
plugins: openai-responses-0.13.1, cov-7.0.0, timeout-2.4.0, anyio-4.12.1, profiling-1.8.1, xdist-3.8.0, jaxtyping-0.3.5, respx-0.22.0, asyncio-0.24.0
timeout: 60.0s
timeout method: signal
timeout func_only: False
session timeout: 480.0s
asyncio: mode=Mode.STRICT, default_loop_scope=None
collected 533 items / 2 errors / 17 deselected / 2 skipped / 516 selected

==================================== ERRORS ====================================
_ ERROR collecting tests/processing/classification/deduplication/test_connected_components.py _
ImportError while importing test module '/home/golem/oa-jder-bot/worktrees/github.com/marin-community/marin/5979813734440518336/tests/processing/classification/deduplication/test_connected_components.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/home/golem/.local/share/uv/python/cpython-3.11.14-linux-aarch64-gnu/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
tests/processing/classification/deduplication/test_connected_components.py:6: in <module>
    from marin.processing.classification.deduplication.connected_components import CCInput, connected_components
lib/marin/src/marin/processing/classification/deduplication/connected_components.py:10: in <module>
    import dupekit
E   ModuleNotFoundError: No module named 'dupekit'
________ ERROR collecting tests/processing/classification/test_decon.py ________
ImportError while importing test module '/home/golem/oa-jder-bot/worktrees/github.com/marin-community/marin/5979813734440518336/tests/processing/classification/test_decon.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/home/golem/.local/share/uv/python/cpython-3.11.14-linux-aarch64-gnu/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
tests/processing/classification/test_decon.py:10: in <module>
    from marin.processing.classification.decon import DeconConfig, DeconMode, NGramConfig, decontaminate
lib/marin/src/marin/processing/classification/decon.py:19: in <module>
    import dupekit
E   ModuleNotFoundError: No module named 'dupekit'
=========================== short test summary info ============================
ERROR tests/processing/classification/deduplication/test_connected_components.py
ERROR tests/processing/classification/test_decon.py
!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!
================= 2 skipped, 17 deselected, 2 errors in 6.69s ================== checks, and reference an issue in the PR body with  or .\n\n## Impact\nReduce avoidable PR churn from missing local checks and missing issue linkage by keeping the contributor-facing guide aligned with current repository practice.

--- a/docs/dev-guide/contributing.md
+++ b/docs/dev-guide/contributing.md
@@ -42,7 +42,11 @@ Before opening a pull request:
 
 1. Run `./infra/pre-commit.py --all-files --fix`.
 2. Run `uv run pytest -m 'not slow'` for the files or packages you changed.
-3. Make sure the PR body references an issue with `Fixes #NNNN` or `Part of #NNNN`.
+3. If your change adds, removes, renames, or rewires docs pages or docs-owned links, run `uv run python infra/check_docs_source_links.py`.
+4. If your change is docs-heavy, run `uv run mkdocs build --strict`.
+5. Keep the PR description concise and plain text because it becomes the squash-merge commit message. The `.github/PULL_REQUEST_TEMPLATE.md` file shows the expected style.
+6. Make sure the PR body references an issue with `Fixes #NNNN` or `Part of #NNNN`.
+7. After pushing, verify the relevant GitHub CI checks pass before considering the PR ready for review.
 
 For the end-to-end branch and fork workflow, follow the PR steps in [Submitting to the Marin Speedrun](../tutorials/submitting-speedrun.md#submit). The same GitHub push and PR process applies to non-speedrun changes.
 

